pylint/checkers/classes/class_checker.py

```diff
@@ -874,6 +881,43 @@ def _check_consistent_mro(self, node: nodes.ClassDef) -> None:
         except astroid.DuplicateBasesError:
             self.add_message("duplicate-bases", args=node.name, node=node)
 
+    def _check_enum_base(self, node: nodes.ClassDef, ancestor: nodes.ClassDef) -> None:
+        members = ancestor.getattr("__members__")
+        if members and isinstance(members[0], nodes.Dict) and members[0].items:
+            self.add_message(
+                "invalid-enum-extension",
+                args=ancestor.name,
+                node=node,
+                confidence=INFERENCE,
+            )
+
+        if ancestor.is_subtype_of("enum.IntFlag"):
+            previous_values, total, union, overlaps = set(), None, None, []
+            for assign_name in node.nodes_of_class(nodes.AssignName):
+                if not isinstance(assign_name.parent, nodes.Assign):
+                    continue  # Ignore non-assignment expressions
+
+                assigned = assign_name.parent.value
+                if not isinstance(assigned, nodes.Const):
+                    continue  # Ignore non-literal assignments
+
+                assert assigned.value is not None
+                if assigned.value in previous_values:
+                    continue  # Ignore aliases
+                previous_values.add(assigned.value)
+
+                # Detect divergence between the sum-of-values and union-of-values
+                total = assigned.value if total is None else total + assigned.value
+                union = assigned.value if union is None else union | assigned.value
+                if total != union:
+                    overlaps.append(assign_name)
+                    total = union  # reset divergence detection after each iteration
+
+            for overlap in overlaps:
+                self.add_message(
+                    "implicit-flag-overlap", node=overlap, confidence=INFERENCE
```

Comment:
I think it would be really nice to use the `args` argument of `add_message` to tell the user a little bit about which flag is overlapping (its name) and potentially also with which other flag.

Reply:
That sounds good - yep, I'll give this a try :+1:

Reply:
Partially-implemented: the message includes information about the flag member where an implicit alias was detected, but doesn't match this up with the corresponding flag(s) that conflict.  I couldn't think of a way to achieve that while also maintaining linear time and space bounding.

Reply:
This is now fully-[implemented](https://github.com/PyCQA/pylint/pull/8105/commits/ac21d3149c2b67365249c16ac5853309de93cea3), hopefully in a reasonably-efficient way.